sdk/src/native_program.rs

```diff
@@ -20,12 +20,32 @@ macro_rules! solana_entrypoint(
     ($entrypoint:ident) => (
         #[no_mangle]
         pub extern "C" fn process(
-            program_id: &Pubkey,
-            keyed_accounts: &mut [KeyedAccount],
+            program_id: &solana_sdk::pubkey::Pubkey,
+            keyed_accounts: &mut [solana_sdk::account::KeyedAccount],
             data: &[u8],
             tick_height: u64
-        ) -> Result<(), InstructionError> {
+        ) -> Result<(), solana_sdk::transaction::InstructionError> {
             $entrypoint(program_id, keyed_accounts, data, tick_height)
         }
     )
 );
+
+// Macro to define an entrypoint from a native `process_instruction` function.
+#[macro_export]
+macro_rules! process_instruction_entrypoint(
```

Comment:
@garious When I look harder at this, it seems to be perfectly duplicating solana_entrypoint()...  I changed vote_program to use solana_entrypoint() directly and suffered no ill effects.  What am I missing?

Reply:
@rob-solana Last time I tried to register the C entrypoint with `Runtime::add_instruction_processor` it complained about a type mismatch.  It's very possible I missed something.